### ISSUE TITLE
Add YaRN scaling to RoPE activations

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#711]
## Problem

Recipe's planned RoPE activation in #710 cannot describe the YaRN frequency scaling stored by models such as RNJ. YaRN belongs to RoPE and must not become an independent model block or attention operation.

## Required API

Document YaRN beneath RoPE:

```rust
.rope(layout, dimensions, base)
	yarn(factor, og_ctx, b_fast, b_slow)
```

The executable builder form is:

```rust
.rope(layout, dimensions, base)
.yarn(factor, og_ctx, b_fast, b_slow)
```

- `factor` is the context extension ratio.
- `og_ctx` is the original training context length.
- `b_fast` is the fast-rotation boundary.
- `b_slow` is the slow-rotation boundary.

YaRN configures the preceding RoPE activation, owns no weights, and is invalid without it. It must preserve all four values through model serialization and GGUF construction.

## Acceptance

- Forward and backward results match a YaRN reference across the extrapolated, blended, and interpolated frequency regions.
- A factor of one matches unscaled RoPE.
- Invalid context, factor, or beta values fail before execution.
- GGUF construction reads `rope.scaling.factor`, `rope.scaling.original_context_length`, `rope.scaling.yarn_beta_fast`, and `rope.scaling.yarn_beta_slow`.
- RNJ metadata values `4`, `8192`, `64`, and `1` match the reference model's first-token logits on CPU and GPU.

Depends on #710.
